File: app/Models/Vehicle.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Vehicle extends Model
{
    use HasFactory;
    
    protected $primaryKey = 'matricula';
    public $incrementing = false;
    protected $keyType = 'string';

    // Incluye 'nombre' junto con los demás campos
    protected $fillable = ['matricula', 'nombre', 'id_parque', 'año', 'tipo'];

    public function vehicles_incident()
    {
        return $this->hasMany('App\Models\Vehicles_incident');
    }

    public function park()
    {
        return $this->belongsTo('App\Models\Park', 'id_parque');
    }
}

File: android/app/src/main/java/es/bomberosgranada/app/data/models/Vehicle.kt
package es.bomberosgranada.app.data.models

data class Vehicle(
    val matricula: String,
    val id_parque: Int,
    val tipo: String,
    val nombre: String,
    val año: Int,
    val created_at: String? = null,
    val updated_at: String? = null,
    // Relación
    val park: Park? = null
)

data class CreateVehicleRequest(
    val matricula: String,
    val nombre: String,
    val id_parque: Int,
    val tipo: String,
    val año: Int
)

data class UpdateVehicleRequest(
    val matricula: String,
    val nombre: String,
    val id_parque: Int,
    val tipo: String,
    val año: Int
)